dotnet/private/common.bzl

```diff
@@ -507,16 +507,36 @@ def framework_preprocessor_symbols(tfm):
     Returns:
         A list of preprocessor symbols.
     """
-    # TODO: All built in preprocessor directives: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives
 
-    specific = tfm.upper().replace(".", "_")
+    # BFS to recursively find all frameworks that are compatible with the target framework
+    compatible_frameworks = []
+    unvisited_frameworks = [tfm]
+    for _ in range(2147483647):
```

Comment:
Can't we find a better way to do this? Not sure what the perf impact during analysis will be if you have a very large repo.

Reply:
Hmm, well, it's either this or DFS, which will be equally unbounded. We could however pre-compute this.

Reply:
I think we can reuse this function here: https://github.com/bazel-contrib/rules_dotnet/blob/51e5e038211106296d6eae99bc755faf0e03ccd3/dotnet/private/common.bzl#L18